spec/rspec/rails/example/view_example_group_spec.rb

```diff
@@ -90,6 +90,27 @@ module ::ApplicationHelper; end
       end
     end
 
+    describe "routes helpers collides with asset helpers" do
+      it "uses routes helpers" do
+        with_isolated_config do
+          ::Rails.application.routes.draw { resources :images }
+          output = nil
+
+          RSpec::Core::ExampleGroup.describe "a view spec", :type => :view do
+            before do
+              controller.assets_dir = "public"
+              allow(controller).to receive(:_routes).
+                and_return(::Rails.application.routes)
+            end
+
+            specify { output = image_path(double(:to_param => "42")) }
```

Comment:
Rather than running a spec within a spec you can then instantiate the class from above and assert this directly